xarray/core/missing.py

```diff
@@ -631,8 +635,15 @@ def interp(var, indexes_coords, method: InterpOptions, **kwargs):
         return var.copy()
 
     result = var
-    # decompose the interpolation into a succession of independent interpolation
-    for indep_indexes_coords in decompose_interp(indexes_coords):
+
+    if reduce:
```

Comment:
```suggestion
    if method in ["linear", "nearest"]:
```

I don't think we need this `reduce` kwarg. It will confuse people.